src/ca/mcgill/cs/jetuml/diagram/ControlFlow.java

```diff
@@ -243,4 +246,124 @@ public boolean isInConstructorCall(Node pNode)
 		}
 		return false;	
 	}
+	
+	/*
+	 * Returns true if pNode is the ImplicitParameterNode that gets created in constructor call
+	 */
+	private boolean isImplicitParameterNodeInConstructorCall(Node pNode)
+	{
+		assert pNode != null;
+		return pNode.getClass() == ImplicitParameterNode.class && pNode.getChildren().size()>0 && isInConstructorCall(getFirstChild(pNode));
+	}
+	
+	private Node getFirstChild(Node pNode)
+	{
+		assert pNode.getChildren().size()>0;
+		return pNode.getChildren().get(0);
+	}
+	
+	private Optional<Edge> getConstructorEdge(Node pNode)
+	{
+		assert pNode != null && isInConstructorCall(pNode);
+		for(Edge edge : aDiagram.edges())
+		{
+			if ( pNode.getClass() == CallNode.class && edge.getEnd() == pNode && edge.getClass() == ConstructorEdge.class )
+			{

```

Comment:
I think `pNode.getClass()` doesn't have to be in the loop, you can move to above and return empty is the condition is true